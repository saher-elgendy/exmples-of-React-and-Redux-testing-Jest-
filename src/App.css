body {
  background: #faf8f8
}

.App {
  text-align: center;
  background: #faf8f8;
}

img {
  max-width: 100%;
  object-fit: contain;
  cursor: pointer;
}

h4 {
  margin: 0.3rem
}

ul {
  padding: 0;
  margin: 0;
  list-style: none;
}

header {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 1.5rem;
}

.invisible-sperator {
  visibility: hidden;
}
.bag-container {
  width: 50px;
  object-fit: contain;
}

.book-count {
  width: 10px;
  height: 10px;
  background: red;
  color: #fff;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 6px
}

.books-list {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
}
.book-card {
  padding: 1rem;
  margin: 1rem;
  width: 250px;
  height: 350px;
  display: flex;
  flex-direction: column;
  border: 2px solid #eee;
  margin-top: 1rem;
  background: #fff;
  flex-grow: 1;
  /* box-shadow: 5px 5px 5px 4px; */
}

.book-cover {
  height: 10rem;
  display: block;
  margin: auto;
}

.book-card-btn {
  display: block;
  border: none;
  padding:  1rem;
  font-weight: bold;
  color: #fff;
  background: dodgerblue;
  cursor: pointer;
  bottom: 1rem;
  margin-top: 0.25rem;
  outline: none;
  transition: background 0.2s ease-in-out;
}

.book-card-btn:hover {
  background: #1b64ac
}

.remove-book-btn {
  background: #0b3e6f
}

.bag-content {
  min-width: 200px;
  min-height: 200px;
  max-width: 80%;
  max-height: 350px;
  overflow: auto;
  background: #fff;
  border: 2px solid #000;
  padding: 0.5rem;
  position: absolute;
  margin:auto
}

.bag-item {
  display: flex;
  align-items: center;
  padding: 0.3rem;
  cursor: pointer;
}

.bag-item:hover {
  background: #eee
}

.bag-book-cover {
  width: 30px;
  margin-right: 10px;
}

.book-details {
  text-align: left
}