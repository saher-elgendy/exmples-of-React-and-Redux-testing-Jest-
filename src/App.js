import React from 'react';
import './App.css';

function App({text}) {
  return (
    <div className="App">
     <header data-testid="header">{text}</header>
    </div>
  );
}

export default App;
