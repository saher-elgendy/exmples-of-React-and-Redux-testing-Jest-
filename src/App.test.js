import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import '@testing-library/jest-dom/extend-expect'

describe('it should', () => {
  test('render App correctly', () => {
    const { container, getByText, getByTestId } = render(<App text="React Testing"></App>);
    //creating snapshot
    expect(container.firstChild).toMatchSnapshot();
    //check if header exist in the document
    const header = getByTestId(/header/);
    expect(header).toHaveTextContent('React Testing');
  })
})