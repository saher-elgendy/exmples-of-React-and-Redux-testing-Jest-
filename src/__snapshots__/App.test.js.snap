// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render App correctly 1`] = `
<div
  class="App"
>
  <header
    data-testid="header"
  >
    React Testing
  </header>
</div>
`;
